js/utility/SaveBtnAction.js

```diff
@@ -0,0 +1,31 @@
+import { CLICK_EVENT, EDIT, COMPLETE, DELETE_TODO } from "/js/constants.js";
+
+export default function saveButtonActionTodo(
+  element,
+  saveBtn,
+  deleteBtn,
+  checkBtn
+) {
+  let removeListener = false;
+
+  function saveButtonAction(event) {
+    event.stopPropagation();
+    event.preventDefault();
+    element.querySelector(".text").contentEditable = false;
+    deleteBtn.style.display = "none";
+    element.querySelector(".saveBtn").style.display = "none";
+    element.querySelector(`[data-job="${DELETE_TODO}"]`).style.display =
+      "inline-block";
+    element.querySelector(`[data-job="${COMPLETE}"]`).style.display =
+      "inline-block";
+    element.querySelector(`[data-job="${EDIT}"]`).style.display =
+      "inline-block";
+    saveBtn.style.display = "none";
+    checkBtn.style.display = "none";
+    removeListener = true;
+  }
+  saveBtn.addEventListener(CLICK_EVENT, saveButtonAction);
+  if (removeListener) {
+    saveBtn.removeEventListener(CLICK_EVENT, saveButtonAction);
+  }
+}
```

Comment:
Can we write the code below and check if it is still working the same as before? If yes can you please identify the difference between the two versions? And if you think writing this way can make the code simpler then you may try for the other two helper file as well. Thank you.

```suggestion
import { CLICK_EVENT, EDIT, COMPLETE, DELETE_TODO } from "/js/constants.js";

export default function handleSaveButtonClick(todoItemElement, saveButtonElement, deleteButtonElement, completeButtonElement) {
  function handleSaveButton(event) {
    event.stopPropagation();
    event.preventDefault();
    todoItemElement.querySelector(".text").contentEditable = false;
    deleteButtonElement.style.display = "none";
    saveButtonElement.style.display = "none";
    todoItemElement.querySelector(`[data-job="${DELETE_TODO}"]`).style.display = "inline-block";
    todoItemElement.querySelector(`[data-job="${COMPLETE}"]`).style.display = "inline-block";
    todoItemElement.querySelector(`[data-job="${EDIT}"]`).style.display = "inline-block";
    completeButtonElement.style.display = "none";
  }
  
  saveButtonElement.addEventListener(CLICK_EVENT, handleSaveButton);
}
```

Reply:
your version of the code enhances the readability of the code.

Reply:
Alright then feel free to mark this conversation as resolved.